controller/login.go

```diff
@@ -112,6 +113,50 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
 	return c.auth.Perform(ctx, oauth, brokerEndpoint, entitlementEndpoint, profileEndpoint, whitelist, c.configuration.GetAuthNotApprovedRedirect())
 }
 
+// getEntitlementResourceRequestPayload creates the object which would have the information about which spaces/resources
+// the entitlements' info would need to be fetched for.
+
+func (c *LoginController) getEntitlementResourceRequestPayload(ctx context.Context, token *string) (*auth.EntitlementResource, error) {
+	loggedInIdentityID, err := c.tokenManager.Extract(*token)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "unable to get ID from access token")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get ID from access token"))
+	}
+
+	// get the user object as well for this identity
+	queryResult, err := c.identityRepository.Query(account.IdentityFilterByID(loggedInIdentityID.ID), account.IdentityWithUser())
+	if err != nil || len(queryResult) == 0 {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err,
+			"identity_id": *loggedInIdentityID,
+		}, "unable to query Identity")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "unable to query Identity"))
+	}
+	loggedInIdentity := queryResult[0]
+	contextInfoLoggedInIdentity := loggedInIdentity.User.ContextInformation
+	if contextInfoLoggedInIdentity == nil {
+		return nil, nil
+	}
+
+	var spacesToGetEntitlementsFor []auth.ResourceSet
+	for _, v := range contextInfoLoggedInIdentity["recentSpaces"].([]interface{}) {
```

Comment:
Agreed, until now the UI was free to change it, but now it gets tied up with the backend a bit - we are safe in a way that if the `recentSpaces` is not found in the contextInformation, the code is going to fall back on the original behaviour - which is to fetch all entitlement info.